### ISSUE TITLE
error during initiation

### ISSUE THREAD

**Issue by ivan-paleo** [rstudio/renv#427]
Dear Kevin,

I am now further than I was a few weeks back, and I have tried for the first time to share my repository to see how it would work with renv.

Unfortunately, not so well...

Here is the error we get directly when opening the Rproj file. Note that this was the second time trying; during the first run, renv tried to download from CRAN (and failed) and then from GitHub (successful) and then installed successfully.

Project 'D:/R/Wor_dir_Rstudio/SSFApaper-master' loaded. [renv 0.9.3-69]  
Error: failed to parse remote 'rstudio/renv@0.9.3-69' -- download failed [error code 35]  
Traceback (most recent calls last):  
37: source("renv/activate.R")  
36: withVisible(eval(ei, envir))  
35: eval(ei, envir)  
34: eval(ei, envir)  
33: local(...) at activate.R#2  
32: eval.parent(substitute(eval(quote(expr), envir)))  
31: eval(expr, p)  
30: eval(expr, p)  
29: eval(quote(...), new.env())  
28: eval(quote(...), new.env())  
27: return(renv::load()) at activate.R#100  
26: renv_load_finish(project, lockfile)  
25: renv_load_report_synchronized(project, lockfile)  
24: renv_project_synchronized_check(project, lockfile)  
23: quietly({         libstate <- snapshot(project = project, library = library,              lockfile = NULL, force = TRUE)         synchronized <- renv_status_check_synchronized(project = project,              lockfile = lockfile, library = library, libstate = libstate)     })  
22: withCallingHandlers(expr, warning = function(c) invokeRestart("muffleWarning"),          message = function(c) invokeRestart("muffleMessage"), packageStartupMessage = function(c) invokeRestart("muffleMessage"))  
21: snapshot(project = project, library = library, lockfile = NULL,          force = TRUE)  
20: renv_lockfile_create(project, library, type)  
19: renv_snapshot_r_packages(library = library, project = project) %>%          renv_snapshot_filter(project = project, type = type) %>%          renv_snapshot_fixup()  
18: eval(call, envir = parent.frame())  
17: eval(call, envir = parent.frame())  
16: renv_snapshot_fixup(renv_snapshot_r_packages(library = library,          project = project) %>% renv_snapshot_filter(project = project,          type = type))  
15: renv_snapshot_fixup_renv(records)  
14: renv_remotes_resolve(remote)  
13: withCallingHandlers(renv_remotes_resolve_impl(entry), error = error)  
12: renv_remotes_resolve_impl(entry)  
11: renv_remotes_resolve_github(parsed)  
10: case(nzchar(pull) ~ renv_remotes_resolve_github_sha_pull(host,          user, repo, pull), nzchar(ref) ~ renv_remotes_resolve_github_sha_ref(host,          user, repo, ref))   
9: eval(expr, envir = environment(dot))   
8: eval(expr, envir = environment(dot))   
7: renv_remotes_resolve_github_sha_ref(host, user, repo, ref)   
6: download(url, destfile = shafile, type = "github", quiet = TRUE,          headers = headers)   
5: stopf("download failed [error code %i]", status)   
4: stop(sprintf(fmt, ...), call. = call.)   
3: .handleSimpleError(function (e)      {         ...     }, "download failed [error code 35]", base::quote(NULL))   
2: h(simpleError(msg, call))   
1: stop(simpleError(message = message, call = e$call))

Any idea what the problem is?

The R version used is somewhat old (3.6.0) but I would not expect that to be a problem.
Here the sessionInfo():
R version 3.6.0 (2019-04-26)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18363)
Matrix products: default
locale:
[1] LC_COLLATE=German_Germany.1252  LC_CTYPE=German_Germany.1252    LC_MONETARY=German_Germany.1252
[4] LC_NUMERIC=C                    LC_TIME=German_Germany.1252    
attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     
loaded via a namespace (and not attached):
[1] compiler_3.6.0 tools_3.6.0    renv_0.9.3-69 

Let me know if you need more info.
Thank you for your help!

**Comment by kevinushey** [comment]
Are you able to update to the latest version of `renv` on CRAN? (0.10.0)

You might also set the option:

```
options(renv.config.synchronized.check = FALSE)
```

in your `.Rprofile`, to disable this startup synchronized check.

**Comment by ivan-paleo** [comment]
I have created the repository with renv 0.9.3-69.
Of course I could update renv and re-initialize renv on my machine before sharing it again with my collaborators. But I thought the point was to make sure that anyone re-running my code will use the same package versions (including renv). 
Or you think there is a bug on that in 0.9.3-69?

**Comment by kevinushey** [comment]
> Of course I could update renv and re-initialize renv on my machine before sharing it again with my collaborators. But I thought the point was to make sure that anyone re-running my code will use the same package versions (including renv).

Right, but if updating `renv` fixes the issue then I'd recommend updating `renv` and also instructing your collaborators to update as well.

The error:

> Error: failed to parse remote 'rstudio/renv@0.9.3-69' -- download failed [error code 35]

indicates something went wrong with the SSL/TLS handshake when `renv` was trying to query information about that version of `renv` from GitHub. See from https://curl.haxx.se/libcurl/c/libcurl-errors.html:

> CURLE_SSL_CONNECT_ERROR (35)
> 
> A problem occurred somewhere in the SSL/TLS handshake. You really want the error buffer and read the message there as it pinpoints the problem slightly more. Could be certificates (file formats, paths, permissions), passwords, and others.

You can try playing with the download method used by `renv` to see if that helps. See https://rstudio.github.io/renv/articles/renv.html#downloads for more details.

**Comment by ivan-paleo** [comment]
OK, now I start to understand the problem; thank you already. So maybe it has nothing to do with renv at then end...
I've updated to the latest version of renv and then we'll try again. I'll let you know.

**Comment by ivan-paleo** [comment]
I tried on another PC and I had no issue. I still haven't managed to check on the PC that threw this error.

**Comment by kevinushey** [comment]
I think ultimately this would be resolved by updating to the latest version of `renv` on CRAN for this project -- feel free to re-open if this does not resolve the issue.

**Comment by ivan-paleo** [comment]
I don't know if I should open a new issue; let me know if I should.

I now tried on a fresh install of R 4.0.0.
It worked fine except for two issues:

1) There is always a problem with package crayon:
* Project 'C:/Users/calandra.RGZM/Downloads/SSFApaper-master' loaded. [renv 0.10.0]
* The project and lockfile are out of sync -- use `renv::status()` for more details.
Warning message:
Project requested R version '3.6.3' but '4.0.0' is currently being used 
Error in loadNamespace(name) : there is no package called ‘crayon’

renv::restore() then installs it and everything is fine, but there is definitely a minor issue here.

2) Some of the packages in my lock file require compilation (I could have avoided it during renv::init() I guess) and Rtools is needed for it. I did not know that R4.0.0 requires a new version of Rtools so some packages could not be compiled and the whole restore() process just stopped with a simple error message like "package could not be complied".
Would it make sense to first check whether Rtools is needed, and if yes, to check whether it is installed?

Thanks,
Ivan
